### DISCUSSION TITLE
Nested Stack Navigators

### DISCUSSION THREAD

**Discussion by elboboua** [nandorojo/solito#82] [Q&A]
Hey! How's it going? I am wondering if you know of a way of nesting stack navigators from react-navigation that can be used by the web app as well as the native app. I have been trying to get createStackNavigator to work on the home screen, but I keep getting Syntax errors unfortunately. Is it possible to do this?
<img width="551" alt="image" src="https://user-images.githubusercontent.com/44094673/173471599-6d6dbb19-0259-4779-a55f-3977a25d55a7.png">

**Comment by nandorojo** [comment]
everything should work as long as 1) the stacks only render on native, 2) you set up your linking config on native such that each URL points to a a screen, and 3) you put those same screens as pages on Next.js

**Reply by nandorojo** [reply]
you shouldn’t use stacks in the home screen, as that would render them on web too. what’s the use case here? do you want a modal on the home screen? i should probably add a modal example

**Comment by dohomi** [comment]
@nandorojo I am bumping into this issue at the moment: I want to split my navigation into several stacks: 
```jsx
const AppStack = createBottomTabNavigator<RouterNavigationProps>()

const ExerciseStack = createNativeStackNavigator<RouterExerciseNavigationProps>()

// part of my navigator is 
                <AppStack.Screen name={'exerciseStack'}
                                 options={{
                                     headerShown: false
                                 }}
                                 component={ExerciseStackNavigation}
                />
```

Now I am not sure how to handle the nesting route, because `screen` is not available in `useLink` based on https://reactnavigation.org/docs/nesting-navigators#navigating-to-a-screen-in-a-nested-navigator

```js
const linkProps = useLink({href: 'exercises'}) 
// => works in NextJS but not in Native because exercises is part of the exerciseStack
```

How can I configure `useLink` to be part of a nested stack? I might just miss something obvious though ..

**Reply by nandorojo** [reply]
you can do this easily! just nest your stacks and then configure nested screens in your linking config.

https://reactnavigation.org/docs/configuring-links/#handling-nested-navigators

**Reply by dohomi** [reply]
Thanks @nandorojo I'll check it out. For now I am using a Group which might even make more sense for my current task
